spiffe-idp-setup-job crash-loops on HyperShift clusters

## Problem

The `kagenti-spiffe-idp-setup-job` pod crash-loops on HyperShift OCP 4.20.11 clusters, causing `test_no_crashlooping_pods` to fail in every HyperShift E2E run. This blocks all HyperShift PR testing.

**Last 5 HyperShift PR runs all fail** at the same test:

```
FAILED tests/e2e/common/test_platform_health.py::TestPlatformHealth::test_no_crashlooping_pods
  AssertionError: Found 1 crashlooping pods:
    kagenti-system/kagenti-spiffe-idp-setup-job-kvqcp
    (container: setup-spiffe-idp, recent restarts: 5, last started: 157s ago)
```

Pod listing at test time (38 min after creation, still Running, not Completed):
```
kagenti-spiffe-idp-setup-job-kvqcp   1/1   Running   5 (4m44s ago)   38m
```

The Ansible installer already logs a warning during deploy:
```
⚠️  Warning: SPIFFE IdP setup job did not complete successfully.
    Check logs with: kubectl logs -n kagenti-system job/kagenti-spiffe-idp-setup-job
```

## Context

- PR #1261 (`fix/spiffe-idp-setup-dockerfile`) fixed the Dockerfile COPY paths so the image can now be built. The `build.yaml` run after that merge succeeded.
- The crash-loop is a **runtime** issue with the script, not a build issue.
- Kind E2E is unaffected (SPIRE may not be enabled on Kind, or the timing is different).

## Where to look

### Job spec (Ansible-managed, not Helm)
- `deployments/ansible/roles/kagenti_installer/tasks/main.yml:1117-1181` — Job definition
- `backoffLimit: 10`, `restartPolicy: OnFailure`
- Init container `wait-for-spire` waits for SPIRE server pod readiness
- Main container `setup-spiffe-idp` runs the Python script

### Script
- `kagenti/auth/spiffe-idp-setup/setup_spiffe_idp.py`
- `wait_for_spire()` — polls `SPIFFE_BUNDLE_ENDPOINT` up to 30 attempts × 10s = 300s
- Then connects to Keycloak to create/update the SPIFFE identity provider
- `KEYCLOAK_TLS_VERIFY` defaults to `true` — verify this is correct for OCP deployments

### Likely failure modes (in order of probability)

1. **SPIRE OIDC Discovery Provider not ready** — the init container only waits for the SPIRE *server* pod, but `wait_for_spire()` checks the OIDC discovery provider endpoint (`http://spire-spiffe-oidc-discovery-provider.spire-server.svc.cluster.local/keys`). If the OIDC provider starts slower than the server on HyperShift, the script exhausts its 30 retries and exits non-zero.

2. **JWKS keys missing `use` field** — if SPIRE's `setKeyUse: true` is not configured, the script returns `False` immediately (line 152-158). Check the SPIRE Helm values for `setKeyUse`.

3. **Keycloak connection failure** — the script connects to `http://keycloak-service.keycloak:8080` (correct for in-cluster). Could fail if Keycloak isn't fully ready, but the Job runs late in the install sequence so this is less likely.

### Key env vars set by Ansible

| Var | Value |
|-----|-------|
| `KEYCLOAK_BASE_URL` | `keycloak.url` from kagenti-deps values |
| `KEYCLOAK_NAMESPACE` | from kagenti-deps values |
| `SPIFFE_TRUST_DOMAIN` | `spiffe://<domain>` |
| `SPIFFE_BUNDLE_ENDPOINT` | `http://spire-spiffe-oidc-discovery-provider.<spire-ns>.svc.cluster.local/keys` |
| `KEYCLOAK_TLS_VERIFY` | **not set** (defaults to `true`) |

### RCA approach

1. Deploy to a HyperShift cluster with `skip_destroy: true`
2. `kubectl logs -n kagenti-system job/kagenti-spiffe-idp-setup-job` to see actual error
3. Check SPIRE OIDC provider readiness: `kubectl get pods -n spire-server`
4. Test the JWKS endpoint manually: `kubectl exec -n kagenti-system <pod> -- curl -s http://spire-spiffe-oidc-discovery-provider.spire-server.svc.cluster.local/keys | jq .`
5. If SPIRE is fine, check Keycloak connectivity from the pod

### CI runs for reference

| Run | Conclusion | URL |
|-----|-----------|-----|
| 24540412500 | failure | https://github.com/kagenti/kagenti/actions/runs/24540412500 |
| 24536365709 | failure | https://github.com/kagenti/kagenti/actions/runs/24536365709 |
| 24534646429 | failure | https://github.com/kagenti/kagenti/actions/runs/24534646429 |
| 24533202420 | failure | https://github.com/kagenti/kagenti/actions/runs/24533202420 |

## Impact

Blocks all HyperShift E2E testing. Kind E2E is unaffected.

--- a/deployments/ansible/roles/kagenti_installer/tasks/main.yml
+++ b/deployments/ansible/roles/kagenti_installer/tasks/main.yml
@@ -838,10 +838,13 @@
 
 #
 #  Patch SPIRE OIDC Discovery Provider ConfigMap
-#  WORKAROUND: The SPIRE helm chart (versions <=0.28.2) does not render the set_key_use
-#  configuration option into the ConfigMap, even though the SPIRE binary supports it.
+#  WORKAROUND: Neither the SPIRE helm chart (versions <=0.28.2) nor the ZTWIM operator (OCP 4.19+)
+#  configures the set_key_use option in the OIDC Discovery Provider ConfigMap.
 #  Keycloak's SPIFFE provider requires "use": "sig" field in JWKS keys.
 #  This task patches the ConfigMap post-install to add set_key_use: true.
+#
+#  NOTE: The SPIFFE IdP setup job (below) depends on this patch. If this block's
+#  `when` condition changes, verify the job condition still aligns. See issue #1267.
 #
 # Patch SPIRE OIDC ConfigMap with set_key_use (must run AFTER kagenti-deps installs SPIRE)
 - name: Patch SPIRE OIDC ConfigMap with set_key_use
@@ -898,34 +901,20 @@
             patched_oidc_config: "{{ spire_oidc_config | combine({'set_key_use': true}) }}"
 
         - name: Apply patched ConfigMap
-          kubernetes.core.k8s:
-            state: present
-            definition:
-              apiVersion: v1
-              kind: ConfigMap
-              metadata:
-                name: spire-spiffe-oidc-discovery-provider
-                namespace: "{{ charts.spire.server_namespace }}"
-                annotations:
-                  meta.helm.sh/release-name: spire
-                  meta.helm.sh/release-namespace: "{{ charts.spire.namespace }}"
-                labels:
-                  app.kubernetes.io/managed-by: Helm
-              data:
-                oidc-discovery-provider.conf: "{{ patched_oidc_config | to_json }}"
-                default.conf: "{{ spire_oidc_cm_data['default.conf'] }}"
-                spiffe-helper.conf: "{{ spire_oidc_cm_data['spiffe-helper.conf'] }}"
+          command: >
+            kubectl patch configmap spire-spiffe-oidc-discovery-provider
+            -n {{ charts.spire.server_namespace }}
+            --type merge
+            -p '{"data":{"oidc-discovery-provider.conf":{{ patched_oidc_config | to_json | to_json }}}}'
           register: oidc_cm_patch_result
 
         - name: Restart SPIRE OIDC Discovery Provider deployment
           command: kubectl rollout restart deployment/spire-spiffe-oidc-discovery-provider -n {{ charts.spire.server_namespace }}
-          when: oidc_cm_patch_result.changed
 
         - name: Wait for SPIRE OIDC Discovery Provider rollout
           command: >
             kubectl rollout status deployment/spire-spiffe-oidc-discovery-provider
             -n {{ charts.spire.server_namespace }} --timeout=2m
-          when: oidc_cm_patch_result.changed
           register: oidc_rollout_result
           retries: 3
           delay: 10
@@ -934,7 +923,6 @@
         - name: Log OIDC ConfigMap patch status
           debug:
             msg: "✅ SPIRE OIDC ConfigMap patched with set_key_use: true and deployment restarted"
-          when: oidc_cm_patch_result.changed
       when:
         - spire_oidc_config is defined
         - not (set_key_use_present | default(false) | bool)
@@ -958,9 +946,16 @@
         msg: "⚠️  Warning: spire-spiffe-oidc-discovery-provider ConfigMap not found. Skipping set_key_use patch."
       when: spire_oidc_cm_check.resources | length == 0
   when: >
-    (charts.spire.enabled | default(false)) and
-    (charts.spire.values['spiffe-oidc-discovery-provider'].enabled | default(true)) and
-    (not (enable_openshift | default(false)) or (use_spire_helm_chart | default(false)))
+    (
+      (charts.spire.enabled | default(false)) and
+      (charts.spire.values['spiffe-oidc-discovery-provider'].enabled | default(true)) and
+      (not (enable_openshift | default(false)) or (use_spire_helm_chart | default(false)))
+    ) or
+    (
+      (enable_openshift | default(false)) and
+      (not (use_spire_helm_chart | default(false))) and
+      (charts['kagenti-deps']['values']['components']['spire']['enabled'] | default(false))
+    )
 
 # Run SPIFFE IdP setup job AFTER set_key_use patch completes
 # This ensures the job doesn't fail due to missing "use" field in JWKS

--- a/kagenti/auth/spiffe-idp-setup/setup_spiffe_idp.py
+++ b/kagenti/auth/spiffe-idp-setup/setup_spiffe_idp.py
@@ -148,13 +148,24 @@ def wait_for_spire(max_attempts: int = 30, delay_seconds: int = 10) -> bool:
                     continue
 
                 # Check if keys have "use" field (required for Keycloak)
+                # Retry instead of failing immediately — the OIDC provider may
+                # still be reloading after a ConfigMap patch + rollout restart.
                 first_key = keys[0]
                 if "use" not in first_key:
-                    logger.error(f"  ❌ JWKS keys missing 'use' field!")
-                    logger.error(f"  SPIRE must be configured with set_key_use: true")
-                    logger.error(
-                        f"  Configure SPIRE OIDC provider with setKeyUse: true in Helm values"
+                    logger.warning(
+                        f"  JWKS keys missing 'use' field (attempt {attempt}/{max_attempts})"
                     )
+                    logger.warning(
+                        f"  SPIRE OIDC provider may still be reloading config"
+                    )
+                    if attempt < max_attempts:
+                        logger.info(f"  Retrying in {delay_seconds} seconds...")
+                        time.sleep(delay_seconds)
+                        continue
+                    logger.error(
+                        f"  ❌ JWKS keys still missing 'use' field after {max_attempts} attempts"
+                    )
+                    logger.error(f"  Ensure SPIRE is configured with set_key_use: true")
                     return False
 
                 logger.info(f"  ✅ SPIRE OIDC Discovery Provider is ready")

--- a/kagenti/tests/e2e/common/test_platform_health.py
+++ b/kagenti/tests/e2e/common/test_platform_health.py
@@ -76,6 +76,9 @@ class TestPlatformHealth:
         Checks that no pods are currently in a CrashLoopBackOff state
         or have recently restarted (within the last 5 minutes).
         Initial startup restarts are ignored.
+        Excludes Job pods since they use restartPolicy: OnFailure and
+        transient restarts are expected — test_all_jobs_completed checks
+        Job completion separately.
         """
         pods = k8s_client.list_pod_for_all_namespaces(watch=False)
 
@@ -84,6 +87,10 @@ class TestPlatformHealth:
         recent_restart_threshold = timedelta(minutes=5)
 
         for pod in pods.items:
+            # Skip Job pods — they restart on failure by design
+            if _is_job_pod(pod):
+                continue
+
             # Check if pod is in CrashLoopBackOff state
             if pod.status.container_statuses:
                 for container in pod.status.container_statuses:

